### PR TITLE
Integrate python server into electron app

### PR DESCRIPTION
**Pull request by pajowu**
- [x] Build python binary using PyOxidizer
- [x] Integrate build binary into electron app
- [x] github action
- [x] Start server from electron app
- [x] implement random port & token selection in server; read them in the client
- [x] pass data dir so started server
- [x] remove this branch from workflow triggers

### CODE REVIEW

**Comment by pajowu** [comment]
Managing & selecting the server still needs some finetuning, I will create a new PR for that after this one is merged.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ on:
   push:
     branches:
       - main
+      - pajowu/pyoxidizer
```
Comment (anuejn) [inline .github/workflows/release.yml:6]: this should go before merging

File: server/build_server.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+set -eu
+
+rm -r app/server &> /dev/null || true
```
Comment (anuejn) [inline server/build_server.sh:4]: maybe rm -rf here?

File: server/build_server.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
```
Comment (anuejn) [inline server/build_server.sh:1]: it is a bit unclear where this script runs maybe define that with something like `cd $(dirname $(realpath $0))`

File: server/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+version: "3.3"
+services:
+  db:
```
Comment (anuejn) [inline server/docker-compose.yml:3]: this whole file seems wyld

File: server/app/main.py [hunk 5]
```diff
@@ -199,17 +231,17 @@ async def start_transcription(
 
 # FIXME: this needs to be removed / put behind proper auth for security reasons
 @app.get("/tasks/list/")
```
Comment (anuejn) [inline server/app/main.py:233]: generally this needs another kind of auth, tho for now it seems to be ok

File: app/src/state/server.ts [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+  }
+  dispatch(setLocalProc(proc));
+  dispatch(setLocalState(LocalServerStatus.Starting));
+  dispatch(addServer({ hostname: 'http://localhost', port: 8000, token: null }));
+  proc.stdout.on('data', (data) => {
+    console.log(`stdout: ${data}`);
+    try {
+      const parsed_data = JSON.parse(data);
+      console.log('parsed_data', parsed_data);
+      if (parsed_data['msg'] == 'server_starting') {
```
Comment (anuejn) [inline app/src/state/server.ts:83]: I think . instread of [''] would be nicer here. also can i haz types?

File: app/src/state/server.ts [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+        dispatch(setToken({ hostname: 'http://localhost', token: parsed_data['token'] }));
+        dispatch(setLocalState(LocalServerStatus.Running));
+      }
+    } catch (e) {
+      console.log('error decoding stdout json', e);
+    }
+  });
+
+  proc.stderr.on('data', (data) => {
+    console.error(`stderr: ${data}`);
```
Comment (anuejn) [inline app/src/state/server.ts:95]: maybe some more meaningfull context like server stderr

File: app/src/components/Controls.tsx [hunk 8]
```diff
@@ -21,3 +24,50 @@ export const Button = styled.button<{ primary?: boolean }>`
       color: var(--bg-color);
     `}
 `;
+
```
Comment (anuejn) [inline app/src/components/Controls.tsx:27]: Is this code pulled in from #30 ?

**Review by anuejn — COMMENTED** [review]
Looks very nice generally, tho I fear rebasing this on #30 or #30 on this ;)

**Comment by pajowu** [comment]
> Looks very nice generally, tho I fear rebasing this on #30 or #30 on this ;)

Well that is my problem now :D

File: server/docker-compose.yml (hunk 4 above)
Comment (pajowu) [reply server/docker-compose.yml:3]: Oops, how did that end up here 🗑️

File: server/app/main.py (hunk 5 above)
Comment (pajowu) [reply server/app/main.py:233]: yup, but i will leave the comment here to remind us to fix that in the future

File: app/src/components/Controls.tsx (hunk 8 above)
Comment (pajowu) [reply app/src/components/Controls.tsx:27]: Yes

**Review by anuejn — APPROVED** [review]

**Pull request merged**